### DISCUSSION TITLE
Detect Schema Change unnecessary to change field data type?

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#829] [🙋♀️TE3 Q&A] [answered]
I had previously thought I needed to do the Detect Schema Change when I was changing the data type of a field - changing the M code or source table's data type.

But today I just edited the M code with the correct data type (changed from Currency.Type to Int64.Type), then for the relevant field, I changed the data type in the properties pane for that field, refreshed the table, and it seemed to hold.

Is this ok, or am I looking at a possible issue down the road?

**Comment by mlonsk** [comment, marked as answer]
To my knowledge you should be fine. Your column now has the property that Detect Schema Change expects and this will not be changed regardless of what the original value was.

**Reply by edhans** [reply]
Thanks. Too much of my code is M and schema changes can be a bit tedious, so for column type changes, this is perfect. I deal with a lot of garbage data and realize during dev when field types are wrong.
